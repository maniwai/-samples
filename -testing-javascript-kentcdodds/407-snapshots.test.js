const data = [
  {name: 'Superman', my_array: ['fly']},
  {name: 'Dash', my_array: ['speed']},
  {name: 'Apogee', my_array: ['fly']},
]

const flyingHeros = () => {
  return data.filter(ele => {
    return ele.my_array.includes('fly')
  })
}
/**
 * Snapshot allow to track data change and update automaticly
 * example: duplicate `fly` will return + 1 array
 */
test('407 | snapshots', () => {
  const data = flyingHeros()
  expect(data).toMatchSnapshot()
})
