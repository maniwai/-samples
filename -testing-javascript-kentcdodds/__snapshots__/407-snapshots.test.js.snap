// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`407 | snapshots 1`] = `
Array [
  Object {
    "my_array": Array [
      "fly",
    ],
    "name": "Superman",
  },
  Object {
    "my_array": Array [
      "fly",
    ],
    "name": "Apogee",
  },
]
`;
